src/NuGet.Indexing/IndexDirectoryProvider/CloudIndexDirectoryProvider.cs

```diff
@@ -75,7 +79,24 @@ public bool Reload(IndexingConfiguration config)
 
             var stopwatch = Stopwatch.StartNew();
 
-            var sourceDirectory = new AzureDirectory(storageAccount, _indexContainerName);
+            AzureDirectory sourceDirectory;
+
+            if (string.IsNullOrEmpty(config.AzureDirectoryCachePath))
+            {
+                sourceDirectory = new AzureDirectory(storageAccount, _indexContainerName);
+            }
+            else
+            {
+                _logger.LogInformation("Using custom azure directory cache path: {AzureDirectoryCachePath}", config.AzureDirectoryCachePath);
+
+                var azureDirectoryCachePath = Path.Combine(config.AzureDirectoryCachePath, _indexContainerName);
+
+                IODirectory.CreateDirectory(azureDirectoryCachePath);
```

Comment:
What will happen if path exists?

Reply:
Should we use `DirectoryInfo.Create` instead?

Reply:
`CreateDirectory` no-ops if the directory already exists.

Reply:
Nothing will happen :)

Source: https://docs.microsoft.com/en-us/dotnet/api/system.io.directory.createdirectory?view=netframework-4.7.1#System_IO_Directory_CreateDirectory_System_String_